roles/openstack_exporter/tasks/main.yml

```diff
@@ -114,3 +114,51 @@
               targetPort: metrics
           selector:
             application: openstack-exporter
+
+- name: Deploy service
+  kubernetes.core.k8s:
+    state: present
+    definition:
+      - apiVersion: apps/v1
+        kind: Deployment
+        metadata:
+          name: openstack-database-exporter
+          namespace: openstack
+          labels:
+            application: openstack-database-exporter
+        spec:
+          replicas: 1
+          selector:
+            matchLabels:
+              application: openstack-database-exporter
+          template:
+            metadata:
+              labels:
+                application: openstack-database-exporter
+            spec:
+              nodeSelector:
+                openstack-control-plane: enabled
+              containers:
+                - name: openstack-database-exporter
+                  image: "{{ atmosphere_images['prometheus_openstack_database_exporter'] | vexxhost.kubernetes.docker_image('ref') }}"
+                  args:
+                    - --neutron.dsn
+                    - "root:{{ openstack_helm_endpoints.oslo_db.auth.admin.password }}@tcp(percona-xtradb-haproxy.openstack.svc.cluster.local:3306)/neutron"  # yamllint disable-line rule:line-length
```

Comment:
I don't really like this, the concern I have with this is that this is a security issue because the password is exposed in the command line.

I'll have to rewrite this to rely on environment variables instead.

Reply:
The new version now can support environment variables to do this, so this should largely be solved.

However, in order to be more neat here, I think what you can do is add the `secretgen-controller` as a dependency to this.

https://github.com/carvel-dev/secretgen-controller/blob/develop/docs/secret-template.md

We can leverage the existing secrets to build the secret so we don't have to run the openstack helm endpoints role, we can just rely on the secrets which are stored in Kubernetes.  It'll be more smart in that sense (and more dynamic).

Once we have that secret, we can just use `envFrom` to get it from that secret.

Reply:
I been searching for a while, but can't find a way to reform secret from 
` mysql+pymysql://neutron:07sj1sB0gjaV4Je5mBIMBpMAewRL4IMT@percona-xtradb-haproxy.openstack.svc.cluster.local:3306/neutron `
to
` neutron:07sj1sB0gjaV4Je5mBIMBpMAewRL4IMT@tcp(percona-xtradb-haproxy.openstack.svc.cluster.local:3306)/neutron `
with SecretTemplate.
I currently using k8s_info to fetch secret, reform it and store back as another secret.